docs/pages/brownfield/integration-with-existing-apps.mdx

```diff
@@ -0,0 +1,654 @@
+
+## Configuring your native project
+
+<Tabs tabs={["Configuring Android", "Configuring iOS"]}>
+<Tab>
+
+To integrate React Native on Android, you need to configure the native project by modifying the following files:
+
+- **Gradle files**: **settings.gradle**, top-level **build.gradle**, **app/build.gradle**, and **gradle.properties** to add the React Native Gradle Plugin (RNGP) and other properties.
+- **AndroidManifest.xml**: To add necessary permissions.
+- **MainActivity**: To load your React Native application.
+
+### Configuring Gradle
+
+<Step label="1">
+Start by editing your **settings.gradle** file and add the following lines (Use the [bare minimum template](https://github.com/expo/expo/blob/main/templates/expo-template-bare-minimum/android/settings.gradle) as a reference):
+
+```groovy settings.gradle
+// Configures the React Native Gradle Settings plugin used for autolinking
+pluginManagement {
+  def reactNativeGradlePlugin = new File(
+    providers.exec {
+      workingDir(rootDir)
+      commandLine("node", "--print", "require.resolve('@react-native/gradle-plugin/package.json', { paths: [require.resolve('react-native/package.json')] })")
+    }.standardOutput.asText.get().trim()
+  ).getParentFile().absolutePath
+  includeBuild(reactNativeGradlePlugin)
+
+  def expoPluginsPath = new File(
+    providers.exec {
+      workingDir(rootDir)
+      commandLine("node", "--print", "require.resolve('expo-modules-autolinking/package.json', { paths: [require.resolve('expo/package.json')] })")
+    }.standardOutput.asText.get().trim(),
+    "../android/expo-gradle-plugin"
+  ).absolutePath
+  includeBuild(expoPluginsPath)
+}
+
+plugins {
+  id("com.facebook.react.settings")
+  id("expo-autolinking-settings")
+}
+
+extensions.configure(com.facebook.react.ReactSettingsExtension) { ex ->
+  ex.autolinkLibrariesFromCommand(expoAutolinking.rnConfigCommand)
+}
+expoAutolinking.useExpoModules()
+
+// rootProject.name = 'HelloWorld'
+
+expoAutolinking.useExpoVersionCatalog()
+
+includeBuild(expoAutolinking.reactNativeGradlePlugin)
+// Include your existing Gradle modules here.
+// include(":app")
+```
+
+If you're using a custom folder structure, you need to explicitly set your project root in **settings.gradle** for autolinking to work. Modify the following lines:
+
+```diff settings.gradle
++expoAutolinking {
++  projectRoot = new File(rootDir, "exp")
++}
+
+extensions.configure(com.facebook.react.ReactSettingsExtension) { ex ->
+-  ex.autolinkLibrariesFromCommand(expoAutolinking.rnConfigCommand)
++  ex.autolinkLibrariesFromCommand(expoAutolinking.rnConfigCommand, file(rootDir), files("yarn.lock"))
+}
+```
+
+</Step>
+
+<Step label="2">
+Then open your top-level **build.gradle** and include this line (as suggested from the [bare minimum template](https://github.com/expo/expo/blob/main/templates/expo-template-bare-minimum/android/build.gradle)):
+
+```diff build.gradle
```

Comment:
can we update these to use our diff components like on https://docs.expo.dev/bare/upgrade/?fromSdk=52&toSdk=53?